File: csharp_body_deref/Scripts/CustomLogic/CustomLogicConditionalBlockAst.cs
// ==================== AoTTG2 cross-reference ====================
// Type: CustomLogic.CustomLogicConditionalBlockAst
// Update status: unchanged since the previous dump
// Real method bodies (native C decompile): source/cpp/CustomLogic/CustomLogicConditionalBlockAst.c
// Prior real C# source (older reference): Assets/Scripts/CustomLogic/Ast/CustomLogicConditionalBlockAst.cs
// Bodies below are empty IL2CPP metadata stubs; the logic lives in the native C.
// ===============================================================
using Il2CppDummyDll;

namespace CustomLogic;

[Token(Token = "0x200028F")]
internal class CustomLogicConditionalBlockAst : CustomLogicBlockAst
{
	[Token(Token = "0x4000DD0")]
	[FieldOffset(Offset = "0x20")]
	public CustomLogicBaseExpressionAst Condition;

	[Token(Token = "0x4000DD1")]
	[FieldOffset(Offset = "0x28")]
	public CustomLogicToken Token;

	[Token(Token = "0x6000F7A")]
	[Address(RVA = "0x3FB8940", Offset = "0x3FB8940", VA = "0x3FB8940")]
	public CustomLogicConditionalBlockAst(CustomLogicToken token, int line)
	{
	}
}

File: cpp/CustomLogic/CustomLogicConditionalBlockAst.c
// Type: CustomLogic.CustomLogicConditionalBlockAst
// Ghidra (System V/gcc) decompilation, IL2CPP structs + signatures applied.
// ---- AoTTG2 cross-reference ----
// Update status: unchanged since the previous dump
// C# structure: source/csharp/Scripts/CustomLogic/CustomLogicConditionalBlockAst.cs
// Prior real C# source (older reference): Assets/Scripts/CustomLogic/Ast/CustomLogicConditionalBlockAst.cs
// --------------------------------

// CustomLogic.CustomLogicConditionalBlockAst$$.ctor
// il2cpp: void CustomLogic_CustomLogicConditionalBlockAst___ctor (CustomLogic_CustomLogicConditionalBlockAst_o* __this, CustomLogic_CustomLogicToken_o* token, int32_t line, const MethodInfo* method);
// 0x40b8940

void CustomLogic_CustomLogicConditionalBlockAst___ctor
               (CustomLogic_CustomLogicConditionalBlockAst_o *__this,CustomLogic_CustomLogicToken_o *token,
               int32_t line,MethodInfo *method)

{
  System_Collections_Generic_List_CustomLogicBaseAst__o *__this_00;
  
  if (g_data_057ac4fb == '\0') {
    il2cpp_runtime_helper_023445d0(&MethodInfo_List_1_CustomLogic_CustomLogicBaseAst);
    il2cpp_runtime_helper_023445d0(&TypeInfo_List_CustomLogicBaseAst);
    g_data_057ac4fb = '\x01';
  }
  __this_00 = (System_Collections_Generic_List_CustomLogicBaseAst__o *)il2cpp_runtime_helper_023052d0(TypeInfo_List_CustomLogicBaseAst);
  System_Collections_Generic_List_object____ctor
            ((System_Collections_Generic_List_object__o *)__this_00,MethodInfo_List_1_CustomLogic_CustomLogicBaseAst);
  (__this->fields).Statements = __this_00;
  il2cpp_runtime_helper_022b4080(&(__this->fields).Statements);
  System_Object___ctor((Il2CppObject *)__this,(MethodInfo *)0x0);
  (__this->fields).Type = 0x10;
  (__this->fields).Line = line;
  (__this->fields).Token = token;
  il2cpp_runtime_helper_022b4080(&(__this->fields).Token,token);
  return;
}


